File: Python Basics/Regular_Exam-27-04-2024/06. Gold Mine/06_gold_mine.py
locations_count = int(input())

for location in range(locations_count):
    expected_average_gold = float(input())
    working_days = int(input())

    location_gr = 0

    for day in range(working_days):
        yield_per_day = float(input())
        location_gr += yield_per_day

    location_average_yield = location_gr / working_days

    if location_average_yield >= expected_average_gold:
        print(f"Good job! Average gold per day: {location_average_yield:.2f}.")
    else:
        diff = expected_average_gold - location_average_yield
        print(f"You need {diff:.2f} gold.")

File: CSharp Development/C# Introduction - Jan 2026/16. Exam/06.GoldMine.cs
using System;

int locationsCount = int.Parse(Console.ReadLine()!);

for (int location = 0; location < locationsCount; location++)
{
    double expectedAverageGold = double.Parse(Console.ReadLine()!);
    int days = int.Parse(Console.ReadLine()!);

    double totalGold = 0.0;

    for (int day = 0; day < days; day++)
    {
        double dailyGold = double.Parse(Console.ReadLine()!);
        totalGold += dailyGold;
    }

    double averageGold = totalGold / days;

    if (averageGold >= expectedAverageGold)
    {
        Console.WriteLine($"Good job! Average gold per day: {averageGold:F2}.");
    }
    else
    {
        double neededGold = expectedAverageGold - averageGold;
        Console.WriteLine($"You need {neededGold:F2} gold.");
    }
}
